packages/edit-site/src/components/page-pages/index.js

```diff
@@ -0,0 +1,190 @@
+/**
+ * WordPress dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+import { addQueryArgs } from '@wordpress/url';
+import {
+	VisuallyHidden,
+	__experimentalHeading as Heading,
+	__experimentalVStack as VStack,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { useEntityRecords } from '@wordpress/core-data';
+import { decodeEntities } from '@wordpress/html-entities';
+import { useState, useEffect, useMemo } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Page from '../page';
+import Link from '../routes/link';
+import PageActions from '../page-actions';
+import { DataViews, PAGE_SIZE_VALUES } from '../dataviews';
+
+const EMPTY_ARRAY = [];
+
+export default function PagePages() {
+	const [ reset, setResetQuery ] = useState( ( v ) => ! v );
+	const [ globalFilter, setGlobalFilter ] = useState( '' );
+	const [ paginationInfo, setPaginationInfo ] = useState();
+	const [ { pageIndex, pageSize }, setPagination ] = useState( {
+		pageIndex: 0,
+		pageSize: PAGE_SIZE_VALUES[ 0 ],
+	} );
+	// Request post statuses to get the proper labels.
+	const [ postStatuses, setPostStatuses ] = useState( EMPTY_ARRAY );
+	useEffect( () => {
+		apiFetch( {
+			path: '/wp/v2/statuses',
```

Comment:
Don't we have an entity for that?

Reply:
No. We could add one in a follow up maybe?

Reply:
Definitely

Reply:
I'm taking this one.

Reply:
https://github.com/WordPress/gutenberg/pull/55050